### PR TITLE
DEV-31474 fix: host header issue, provide raw response json

### PR DESCRIPTION
**Pull request by abhijeetnarvekar**
## Fixes

- Fix host header
-  Migrate to `react\http` from `clue/reactphp-buzz` as its deprecated.
- Allow raw response JSON to be made available via Check Result
- Do not parse custom fields
- Update GH action yaml to support manual triggers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ Our [tests](tests/AcrolinxEndpointTest.php) cover a lot of use cases already. Us
 
 * rename the `.env.conf`file to `.env`
 * set `ACROLINX_TEST_SERVER_URL` and `ACROLINX_ACCESS_TOKEN` in this file. ([Get an access token](https://docs.acrolinx.com/cli/latest/en/how-to/get-an-access-token))
+* set `set ACROLINX_SSO_USER` and `ACROLINX_SSO_PASSWORD` (You can request the SSO password from Acrolinx support)
```
Comment (loaiabdelhalim) [inline README.md]: Simple typo here

File: src/AcrolinxEndpoint.php [hunk 2]
```diff
@@ -106,11 +106,6 @@ private function getCommonHeaders($authToken)
             $headers['X-Acrolinx-Auth'] = $authToken;
         }
 
-        if(isset($_SERVER['HTTP_HOST'])) {
```
Comment (loaiabdelhalim) [inline src/AcrolinxEndpoint.php:109]: What's the reason behind removing this check?

Comment (abhijeetnarvekar) [reply src/AcrolinxEndpoint.php:109]: The react HTTP client sets it automatically for us, based on the request domain.

_Note that the server supports any request method (including custom and non- standard ones) and all request-target formats defined in the HTTP specs for each respective method, including normal origin-form requests as well as proxy requests in absolute-form and authority-form. The getUri(): UriInterface method can be used to get the effective request URI which provides you access to individiual URI components. Note that (depending on the given request-target) certain URI components may or may not be present, for example the getPath(): string method will return an empty string for requests in asterisk-form or authority-form. Its getHost(): string method will return the host as determined by the effective request URI, which defaults to the local socket address if an HTTP/1.0 client did not specify one (i.e. no Host header). Its getScheme(): string method will return http or https depending on whether the request was made over a secure TLS connection to the target host.
The Host header value will be sanitized to match this host component plus the port component only if it is non-standard for this URI scheme.
You can use getMethod(): string and getRequestTarget(): string to check this is an accepted request and may want to reject other requests with an appropriate error code, such as 400 (Bad Request) or 405 (Method Not Allowed)._

Refer: https://github.com/reactphp/http#request-method

File: README.md (hunk 1 above)
Comment (abhijeetnarvekar) [reply README.md]: Removed duplicate "set"

**Pull request merged**